### ISSUE TITLE
fn:subsequence-where: equivalent `fn:slice` expression

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#1001]
I probably should have tagged #940 with »Request Changes«, as I believe the equivalent expression with `fn:slice` needs to be fixed (or removed if it turns out to be too quirky): https://github.com/qt4cg/qtspecs/pull/940#issuecomment-1919399348.

**Comment by michaelhkay** [comment]
I'm trying to get to the bottom of this.

First problem I think is that `otherwise` binds more tightly than `+`, so it should be 

`let $start := index-where($input, $from)[1] 
              otherwise (count($input) + 1)`

I'm then getting a problem that in test subsequence-before-002, the expression `mock:subsequence-where($in, to:=$pred)` returns a different result from `mock:subsequence-where($in, from:=true#0, to:=$pred)`. That must be a bug in Saxon's handling of default function parameters.

**Comment by michaelhkay** [comment]
Most of the `subsequence-where` tests are now passing using a mock implementation based on the equivalence expression in the spec, with parentheses added as described above.

The exceptions are tests subsequence-after-001, and subsequence-ending-where-101 to -108.

Those turned out to be errors in the way I converted the tests to call the mock version of the function. With that fixed, all the tests are now producing the expected results using the function:

```
declare function mock:subsequence-where (
  $input	as item()*,	
  $from	as (function(item(), xs:integer) as xs:boolean)?	:= true#0,
  $to	as (function(item(), xs:integer) as xs:boolean)?	:= false#0
) as item()*
{
   let $start := trace(index-where($input, $from)[1] 
              otherwise (count($input) + 1), '$start')
   let $end :=   trace(index-where($input, $to)[. ge $start][1] 
              otherwise (count($input) + 1), '$end')
   return slice($input, $start, $end)
};

```
As this exercise found a couple of Saxon bugs, I'm inclined to keep the tests for posterity...

**Comment by michaelhkay** [comment]
Fixed by PR #1025
